src/payments-methods/express/payment-methods-express.js

```diff
@@ -14,19 +14,48 @@ import { renderAmazonButton } from '../../renderAmazonButton';
  * @returns React component
  */
 const AmazonPayExpressBtn = ( props ) => {
+	const estimatedOrderAmount = calculateEstimatedOrderAmount( props );
+
 	useEffect( () => {
-		renderAmazonButton( '#pay_with_amazon_express', 'express', null );
+		renderAmazonButton( '#pay_with_amazon_express', 'express', null, estimatedOrderAmount );
 	}, [] );
 
 	return <div id="pay_with_amazon_express" />;
 };
 
+/**
+ * Returns the estimated order amount button attribute.
+ * @param {object} props
+ * @returns {object}|null Estimated order amount button attribute.
+ */
+const calculateEstimatedOrderAmount = ( props ) => {
+	const { billing } = props;
+	const { currency } = billing;
+
+	const stringCartTotal     = String( billing.cartTotal.value );
+	const cartTotalLength     = stringCartTotal.length;
+	const decimals            = currency.minorUnit;
+	const checkOutValue       = stringCartTotal.slice( 0, cartTotalLength - decimals ) + '.' + stringCartTotal.slice( cartTotalLength - decimals );
+
+	return cartTotalLength < decimals ? null : { amount: checkOutValue, currencyCode: currency.code };
+};
+
 /**
  * Returns the Components that will be used by Amazon Pay "Express".
  *
  * @param {object} props
  * @returns React Component
  */
 export const AmazonExpressContent = ( props ) => {
-	return <AmazonPayExpressBtn { ...props } />;
+	const estimatedOrderAmount = calculateEstimatedOrderAmount( props );
+
+	const [id, setId] = useState( estimatedOrderAmount ? estimatedOrderAmount.amount + estimatedOrderAmount.currencyCode : '0' );
```

Comment:
Could we do something like that, for better clarity?
```
const key = estimatedOrderAmount ? `${ estimatedOrderAmount.amount }${ estimatedOrderAmount.currencyCode }` : '0' 
```

Reply:
Yes of course, amended!